apps/ledger-live-desktop/src/newArch/features/ModularDrawer/components/SelectAsset.tsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { CryptoOrTokenCurrency } from "@ledgerhq/types-cryptoassets";
+import { AssetList, AssetType } from "@ledgerhq/react-ui/pre-ldls";
+import { track } from "~/renderer/analytics/segment";
+import TrackPage from "~/renderer/analytics/TrackPage";
+import { Box, Flex } from "@ledgerhq/react-ui/index";
+import { Skeleton } from "LLD/components/Skeleton";
+
+type SelectAssetProps = {
+  assetTypes?: AssetType[];
+  assetsToDisplay: CryptoOrTokenCurrency[];
+  source: string;
+  flow: string;
+  onAssetSelected: (asset: CryptoOrTokenCurrency) => void;
+};
+
+const CURRENT_PAGE = "Modular Asset Selection";
+const ROW_HEIGHT = 64;
+const SPACING = 8;
+const TWO_ROWS_HEIGHT = 2 * ROW_HEIGHT;
+const INPUT_HEIGHT = 40;
+const EXTRA_BOTTOM_MARGIN = TWO_ROWS_HEIGHT + INPUT_HEIGHT + SPACING;
+const EXTRA_PADDING_HIDDEN_SCROLLBAR = 12;
+
+export const SelectAsset = ({
+  assetTypes,
+  assetsToDisplay,
+  source,
+  flow,
+  onAssetSelected,
+}: SelectAssetProps) => {
+  const shouldDisplayLoading = !assetTypes || assetTypes.length === 0;
+
+  const onClick = (asset: AssetType) => {
```

Comment:
I think it's worth wrapping this in a useCallback to avoid unnecessary `AssetList` re-renders